napari/utils/colormaps/standardize_color.py

```diff
@@ -0,0 +1,331 @@
+    Nx4 np.float32 values, with N being the number of colors. The function
+    (via its subfunctions, marked with _handle_X) is designed to parse all
+    valid color representations a user might have and convert them properly.
+    That being said, combinations of different color representation in the
+    same list of colors is prohibited, and will error. This means that a list
+    of ['red', np.array([1, 0, 0])] cannot be parsed and has to be manually
+    pre-processed by the user before sent to this function.
+
+    Parameters
+    --------
+    colors : string, array-like, Color and ColorArray instances, or a mix of the above.
+        The color(s) to interpret and convert
+
+    Returns
+    ------
+    colors : np.ndarray
+        An instance of np.ndarray with a data type of float32, 4 columns in
+        RGBA order and N rows, with N being the number of colors. The array
+        will always be 2D even if a single color is passed.
+
+    Raises
+    -----
+    ValueError, AttributeError, KeyError
+        invalid inputs
+    """
+    colortype = type(colors)
+    return _color_switch[colortype](colors)
+
+
+@functools.lru_cache(maxsize=1024)
+def _handle_str(color: str) -> np.ndarray:
+    """Creates an array from a color that was given as a string."""
+    # This line will stay here until vispy adds a "transparent" key
+    # to their color dictionary. A PR was sent and approved, currently
+    # waiting to be merged.
+    if len(color) == 0:
+        warnings.warn(
+            "Empty string detected. Returning a black color instead."
+        )
+        return np.zeros((1, 4), dtype=np.float32)
+    color = color.replace("transparent", "#00000000")
+    as_arr = np.atleast_2d(_string_to_rgb(color)).astype(np.float32)
+    if as_arr.shape[1] == 3:
+        as_arr = np.column_stack([as_arr, np.float32(1.0)])
+    return as_arr
+
+
+def _handle_list_like(colors: Iterable) -> np.ndarray:
+    """Handles all list-like containers of colors, using recursion. Numpy
+    arrays are handled in _handle_array. Lists which are known to contain
+    strings will be parsed with _handle_str_list_like.
+    """
+    try:
+        # The following conversion works for most cases, and so it's expected
+        # that most valid inputs will pass this .asarray() call
+        # with ease. Those who don't are usually too cryptic to decipher. The
+        # only exception is a list-like container with Vispy's Color and
+        # ColorArray which is considered valid.
+        color_array = np.atleast_2d(np.asarray(colors))
+    except ValueError:
+        if type(colors[0]) in (Color, ColorArray):
+            color_array = np.vstack([c.rgba for c in colors])
+            return color_array
+        else:
+            warnings.warn(
+                "Coudln't convert input color array to a proper numpy array."
+                " Please make sure that your input data is in a parsable format."
```

Comment:
"A parsable format" would be easier to interpret with a Parameters description in the docstring!